include/dali/core/geom/geom_utils.h

```diff
@@ -46,6 +46,21 @@ mat<N, M, T> as_mat(TensorView<StorageCPU, const T, DynamicDimensions> view) {
   return as_mat<N, M, T>(view.template to_static<2>());
 }
 
+template <int N, typename T, typename U>
+void strides(vec<N, U>& out, U& total_stride, const vec<N, T>& v) {
+  total_stride = 1;
+  for (int d = 0; d < N; d++) {
+    out[d] = total_stride;
+    total_stride *= v[d];
+  }
+}
+
+template <int N, typename T, typename U>
+void strides(vec<N, U>& out, const vec<N, T>& v) {
+  U total_strides;
+  strides(out, total_strides, v);
+}
```

Comment:
How about:
```suggestion
template <int N, typename T, typename U>
U strides(vec<N, U>& out, const vec<N, T>& v) {
  U total_stride = 1;
  for (int d = 0; d < N; d++) {
    out[d] = total_stride;
    total_stride *= v[d];
  }
  return total_stride;
}
```

Also: I'm not convinced this is the correct file to put this utility in. This could land in the same place that we have CalcStrides - and, indeed, we could extend CalcStrides to support different traversal directions, e.g.:
```C++
template <bool outer_first = true, typename Strides, typename Shape>
auto CalcStrides(Strides &strides, const Shape& shape) {
  using stride_t = std::remove_reference_t<decltype(strides[0])>;
  int ndim = dali::size(shape);
  resize_if_possible(strides, ndim);  // no-op if strides is a plain array or std::array
  stride_t ret = 1;
  if (outer_first) {
    for (int d = ndim - 1; d > 0; d--) {
      strides[d] = ret;
      ret *= shape[d];
    }
  } else {
    for (int d = 0; d < ndim; d++) {
      strides[d] = ret;
      ret *= shape[d];
    }
  }
  return ret;
}
```

Then you could use it like:
```C++
total_stride = CalcStrides<false>(strides, shape);
```

Reply:
Sounds good, I'll move it.

Reply:
CalcStrides uses int64_t for strides accumulation and it likely needs to stay that way, as the strides[0] can, for example in transpoge_gpu_impl, can be fast_div<uint64_t>, that is not happy about `*=` operator. I left it that way:

```
template <bool outer_first = true, typename Strides, typename Shape>
DALI_HOST_DEV std::remove_reference_t<decltype(std::declval<Strides>()[0])> CalcStrides(
    Strides &strides, const Shape &shape) {
  int ndim = dali::size(shape);
  resize_if_possible(strides, ndim);  // no-op if strides is a plain array or std::array
  int64_t ret = 1;
  if (outer_first) {
    for (int d = ndim - 1; d >= 0; d--) {
      strides[d] = ret;
      ret *= shape[d];
    }
  } else {
    for (int d = 0; d < ndim; d++) {
      strides[d] = ret;
      ret *= shape[d];
    }
  }
  return ret;
}
```